syne_tune/blackbox_repository/conversion_scripts/scripts/pd1_import.py

```diff
@@ -0,0 +1,315 @@
+        bb.hyperparameters.to_parquet(
+            path / f"{task}-hyperparameters.parquet",
+            index=False,
+            compression="gzip",
+            engine="fastparquet",
+        )
+
+        with open(path / f"{task}-fidelity_space.json", "w") as f:
+            json.dump(
+                {
+                    k: config_space.to_dict(v)
+                    for k, v in bb_dict[task].fidelity_space.items()
+                },
+                f,
+            )
+
+        with open(path / f"{task}-objectives_evaluations.npy", "wb") as f:
+            np.save(
+                f,
+                bb_dict[task].objectives_evaluations.astype(np.float32),
+                allow_pickle=False,
+            )
+
+        with open(path / f"{task}-fidelity_values.npy", "wb") as f:
+            np.save(f, bb_dict[task].fidelity_values, allow_pickle=False)
+
+    metadata = metadata.copy() if metadata else {}
+    metadata.update(
+        {
+            "objectives_names": bb_first.objectives_names,
+            "task_names": list(bb_dict.keys()),
+        }
+    )
+    serialize_metadata(
+        path=path,
+        metadata=metadata,
+    )
+
+
+def deserialize(path: str) -> Dict[str, BlackboxTabular]:
+    """
+    Deserialize blackboxes contained in a path that were saved with `serialize` above.
+    TODO: the API is currently dissonant with `serialize`, `deserialize` for BlackboxOffline as `serialize` is there a member.
+    A possible way to unify is to have serialize also be a free function for BlackboxOffline.
+    :param path: a path that contains blackboxes that were saved with `serialize`
+    :return: a dictionary from task name to blackbox
+    """
+    path = Path(path)
+    configuration_space, _ = deserialize_configspace(path)
+    metadata = deserialize_metadata(path)
+    objectives_names = metadata["objectives_names"]
+    task_names = metadata["task_names"]
+
+    bb_dict = {}
+    for task in task_names:
+        hyperparameters = pd.read_parquet(
+            Path(path) / f"{task}-hyperparameters.parquet", engine="fastparquet"
+        )
+        with open(path / f"{task}-fidelity_space.json", "r") as file:
+            fidelity_space = {
+                k: config_space.from_dict(v) for k, v in json.load(file).items()
+            }
+
+        with open(path / f"{task}-fidelity_values.npy", "rb") as f:
+            fidelity_values = np.load(f)
+
+        with open(path / f"{task}-objectives_evaluations.npy", "rb") as f:
+            objectives_evaluations = np.load(f)
+
+        bb_dict[task] = BlackboxTabular(
+            hyperparameters=hyperparameters,
+            configuration_space=configuration_space,
+            fidelity_space=fidelity_space,
+            objectives_evaluations=objectives_evaluations,
+            fidelity_values=fidelity_values,
+            objectives_names=objectives_names,
+        )
+    return bb_dict
+
+
+if __name__ == "__main__":
+    PD1Recipe().generate()
```

Comment:
What checks did you do? Did you try to plot the learning-curves and run a simulation?

Reply:
I've plotted some learning curves, manually spot-checked raw and transformed data and added the benchmark to the AutoML snippet and ran ASHA on it for one task only.